testsuite/MDAnalysisTests/analysis/test_hole2.py

```diff
@@ -294,6 +294,19 @@ def test_sph_process_oserror(self, universe):
             h = hole2.HoleAnalysis(universe, sph_process='dummy_path')
 
 
+class TestHoleNoneFilename(BaseTestHole):
+
+    @pytest.fixture()
+    def universe_none_filename(self):
+        u = mda.Universe(PDB_HOLE, in_memory=True)
+        u.trajectory.filename = None
+        return u
+
+    def test_none_filename(self, universe_none_filename, tmpdir):
+        with tmpdir.as_cwd():
+            with hole2.HoleAnalysis(universe_none_filename) as h:
+                h.run()
+
```

Comment:
As this is just a single test of HoleAnalysis, could you please just keep it as a single function in the TestHoleAnalysis class? The `universe_none_filename` fixture is only used for this one test, so it could just get created within the function itself.

Reply:
Yes you're right, that would be better